library/src/main/java/no/finntech/capturandro/OrientationUtil.kt

```diff
@@ -0,0 +1,77 @@
+package no.finntech.capturandro
+
+import android.content.ContentResolver
+import android.net.Uri
+import android.provider.MediaStore
+import android.util.Log
+import androidx.exifinterface.media.ExifInterface
+import java.io.File
+import java.io.IOException
+
+object OrientationUtil {
+
+    @JvmStatic
+    fun getOrientation(photoUri: Uri, contentResolver: ContentResolver): Int {
+        val exif = readExif(photoUri, contentResolver)
+        if (exif != null) {
+            return getOrientation(exif)
+        } else {
+            val projection = arrayOf(MediaStore.Images.ImageColumns.ORIENTATION)
+            val cursor = contentResolver.query(photoUri, projection, null, null, null)
+            if (cursor != null) {
+                if (cursor.moveToFirst()) {
+                    return cursor.getInt(0)
```

Comment:
I actually didn't do anything to guard against the `CursorIndexOutOfBoundsException` here. Still not sure how it could happen 🤷‍♂️

Reply:
I mean, I guess you _could_ do `&& cursor.getColumnCount() > 0` in the condition, but I agree it shouldn't be necessary.